cli/cmd/edges.go

```diff
@@ -0,0 +1,297 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/linkerd/linkerd2/controller/api/util"
+	pb "github.com/linkerd/linkerd2/controller/gen/public"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type indexedEdgeResults struct {
+	ix   int
+	rows []*pb.Edge
+	err  error
+}
+
+func newCmdEdges() *cobra.Command {
+	options := newStatOptions()
```

Comment:
we might want to consider using a new `edgesOptions` struct:
```go
type edgesOptions struct {
	namespace    string
	outputFormat string
}
```

`statOptions` supports a number of fields that are unused by `edges`. the underlying `statOptionsBase` is close, but it's still a superset of what we need. alternatively, you could create a struct underneath `statOptionsBase`:
```go
type optionsBase struct {
	namespace    string
	outputFormat string
}
type statOptionsBase struct {
    optionsBase
	timeWindow   string
}
```

another motivation for not re-using `statOptions` is that the output format validation will succeed on `-owide` even though `edges` doesn't support it:
```bash
$ bin/linkerd edges -n linkerd deploy -owide
panic: runtime error: slice bounds out of range
```

Reply:
I was attempting to re-use `statOptions` in order to DRY things up but you're right, the validation error is 👎 . Good point.